lib/vdc/DCBOV.cpp

```diff
@@ -0,0 +1,398 @@
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <iostream>
+#include <fstream>
+#include "vapor/VAssert.h"
+#include <stdio.h>
+
+#ifdef _WINDOWS
+    #define _USE_MATH_DEFINES
+    #pragma warning(disable : 4251 4100)
+#endif
+#include <cmath>
+
+#include <vapor/GeoUtil.h>
+#include <vapor/UDUnitsClass.h>
+#include <vapor/DCBOV.h>
+#include <vapor/DCUtils.h>
+
+using namespace VAPoR;
+using namespace std;
+
+DCBOV::DCBOV() : _bovCollection(nullptr)
+{
+    _dimsMap.clear();
+    _coordVarsMap.clear();
+    _dataVarsMap.clear();
+    _meshMap.clear();
+    _coordVarKeys.clear();
+    _derivedVars.clear();
+}
+
+DCBOV::~DCBOV()
+{
+    for (int i = 0; i < _derivedVars.size(); i++) {
+        if (_derivedVars[i]) delete _derivedVars[i];
+    }
+    _derivedVars.clear();
+
+    if (_bovCollection != nullptr) { delete _bovCollection; }
+}
+
+int DCBOV::initialize(const vector<string> &paths, const std::vector<string> &options)
+{
+    _bovCollection = new BOVCollection();
+    int rc = _bovCollection->Initialize(paths);
+    if (rc < 0) {
+        SetErrMsg("Failure reading .bov file");
+        return (-1);
+    }
+
+    //
+    //  Get the dimensions of the grid.
+    //	Initializes members: _dimsMap
+    //
+    rc = _InitDimensions();
+    if (rc < 0) {
+        SetErrMsg("No valid dimensions");
+        return (-1);
+    }
+
+    // Set up the coordinate variables
+    //
+    rc = _InitCoordinates();
+    if (rc < 0) { return (-1); }
+
+    // Identify data and coordinate variables. Sets up members:
+    // Initializes members: _dataVarsMap, _meshMap
+    //
+    rc = _InitVars();
+    if (rc < 0) return (-1);
+
+    return (0);
+}
+
+int DCBOV::_InitDimensions()
+{
+    _dimsMap.clear();
+    std::vector<std::string> dimnames = _bovCollection->GetSpatialDimensions();
+    std::vector<size_t>      dimlens = _bovCollection->GetDataSize();
+    VAssert(dimnames.size() == 3);
```

Comment:
Does the BOV format not support 2D data?

Reply:
3 dimensions need to be described in each BOV file.  A 2D variable could be described by assigning the dimensionality of one axis as 1.  

IE:
   10x10x1